app/src/org/commcare/android/tasks/TemplatePrinterTask.java

```diff
@@ -0,0 +1,253 @@
+package org.commcare.android.tasks;
+
+import android.os.AsyncTask;
+import android.os.Bundle;
+
+import java.io.File;
+import java.io.IOException;
+
+import org.commcare.android.util.TemplatePrinterUtils;
+
+
+/**
+ * Asynchronous task for populating an html document with data.
+ *
+ * @author Richard Lu
+ * @author amstone
+ */
+public class TemplatePrinterTask extends AsyncTask<Void, Void, Integer> {
+
+    /**
+     * The 4 result codes that can be sent back by this task
+     */
+    public static final int SUCCESS = 0;
+    public static final int IO_ERROR = 1;
+    public static final int VALIDATION_ERROR_MUSTACHE = 2;
+    public static final int VALIDATION_ERROR_CHEVRON = 3;
+
+    /**
+     * Used to track which type of error (2 or 3 above) was encountered in validateString
+     */
+    private static int validationErrorType;
```

Comment:
Is there a reason this is static? If there are ever 2 async template tasks running at once, this will be problematic.
If you refactor exception raising/handling (I'll point out a good way in a bit) you probably don't even need this field.